sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/Shard.java

```diff
@@ -0,0 +1,97 @@
+        ChangefeedCursor userCursor, ChunkFactory chunkFactory) {
+        this.client = client;
+        this.shardPath = shardPath;
+        this.segmentCursor = segmentCursor;
+        this.userCursor = userCursor;
+        this.chunkFactory = chunkFactory;
+    }
+
+    /**
+     * Get events for the Shard.
+     * @return A reactive stream of {@link BlobChangefeedEventWrapper}
+     */
+    Flux<BlobChangefeedEventWrapper> getEvents() {
+        /* List relevant chunks. */
+        return listChunks()
+            .concatMap(chunkPath -> {
+                /* Defaults for blockOffset and objectBlockIndex. */
+                long blockOffset = 0;
+                long objectBlockIndex = 0;
+                /* If a user cursor was provided and it points to this chunk path, the chunk should get events based
+                   off the blockOffset and objectBlockIndex.
+                   This just makes sure only the targeted chunkPath uses the blockOffset and objectBlockIndex to
+                   read events. Any subsequent chunk will read all of its events. */
+                if (userCursor != null && userCursor.getChunkPath().equals(chunkPath)) {
+                    blockOffset = userCursor.getBlockOffset();
+                    objectBlockIndex = userCursor.getObjectBlockIndex();
+                }
+                return chunkFactory.getChunk(client, chunkPath, segmentCursor.toChunkCursor(chunkPath),
+                    blockOffset, objectBlockIndex)
+                    .getEvents();
+            });
+    }
+
+    /**
+     * Lists relevant chunks in a shard.
+     * @return A reactive stream of chunks.
+     */
+    private Flux<String> listChunks() {
+        Flux<String> chunks = client.listBlobs(new ListBlobsOptions().setPrefix(shardPath))
+            .map(BlobItem::getName);
+        /* If no user cursor was provided, just return all chunks without filtering. */
+        if (userCursor == null) {
+            return chunks;
+        /* If a user cursor was provided, filter out chunks that come before the chunk specified in the cursor. */
+        } else {
+            AtomicBoolean pass = new AtomicBoolean(); /* Whether or not to pass the event through. */
+            return chunks.filter(chunkPath -> {
+                if (pass.get()) {
+                    return true;
+                } else {
+                    /* If we hit the chunk specified in the user cursor, set pass to true and pass this chunk
+                       and any subsequent chunks through. */
+                    if (userCursor.getChunkPath().equals(chunkPath)) {
+                        pass.set(true); /* This allows us to pass subsequent chunks through.*/
+                        return true; /* This allows us to pass this chunk through. */
+                    } else {
+                        return false;
+                    }
+                }
+            });
```

Comment:
Let me know if I understand this incorrectly.

So the structure of these paths would be similar to the following

```
"$blobchangefeed/log/00/2019/02/22/1810/",
"$blobchangefeed/log/01/2019/02/22/1810/"
```
This is checking if we hit a chunk with a matching cursor to what a customer passed such as `2019/02`.

Could this logic be simplified, with a small overhead in perf, to just check `chunkPath.startsWith(userCursor.getChunkPath())` and that be the only logic needed to be checked.

Reply:
So shards will be of the form $blobchangefeed/log/00/2019/02/22/1810/, $blobchangefeed/log/01/2019/02/22/1810/

And the chunks under them will be of the form  
$blobchangefeed/log/00/2019/02/22/1810/0000.avro,
$blobchangefeed/log/00/2019/02/22/1810/0001.avro,  $blobchangefeed/log/00/2019/02/22/1810/0002.avro, etc

The user cursor will maybe have stored something that looks like chunkPath = $blobchangefeed/log/00/2019/02/22/1810/0001.avro.

and ideally the filter will return $blobchangefeed/log/00/2019/02/22/1810/0001.avro, and $blobchangefeed/log/00/2019/02/22/1810/0002.avro

So I dont think startsWith would be too applicable here.

Reply:
I'll add a comment with the structure of the paths